packages/next-urql/src/with-urql-client.tsx

```diff
@@ -50,54 +51,58 @@ export function withUrqlClient(
     // Set the displayName to indicate use of withUrqlClient.
     withUrql.displayName = `withUrqlClient(${getDisplayName(AppOrPage)})`;
 
-    withUrql.getInitialProps = async (appOrPageCtx: NextUrqlContext) => {
-      const { AppTree } = appOrPageCtx;
-
-      // Determine if we are wrapping an App component or a Page component.
-      const isApp = !!(appOrPageCtx as AppContext).Component;
-      const ctx = isApp
-        ? (appOrPageCtx as AppContext).ctx
-        : (appOrPageCtx as NextPageContext);
-
-      const opts =
-        typeof clientConfig === 'function' ? clientConfig(ctx) : clientConfig;
-      const [urqlClient, ssrCache] = initUrqlClient(opts, mergeExchanges);
-
-      if (urqlClient) {
-        (ctx as NextUrqlContext).urqlClient = urqlClient;
-      }
-
-      // Run the wrapped component's getInitialProps function.
-      let pageProps = {} as any;
-      if (AppOrPage.getInitialProps) {
-        pageProps = await AppOrPage.getInitialProps(appOrPageCtx as any);
-      }
-
-      // Check the window object to determine whether or not we are on the server.
-      // getInitialProps runs on the server for initial render, and on the client for navigation.
-      // We only want to run the prepass step on the server.
-      if (typeof window !== 'undefined') {
-        return { ...pageProps, urqlClient };
-      }
-
-      const props = { ...pageProps, urqlClient };
-      const appTreeProps = isApp ? props : { pageProps: props };
-
-      // Run the prepass step on AppTree. This will run all urql queries on the server.
-      await ssrPrepass(<AppTree {...appTreeProps} />);
-
-      // Serialize the urqlClient to null on the client-side.
-      // This ensures we don't share client and server instances of the urqlClient.
-      (urqlClient as any).toJSON = () => {
-        return null;
-      };
+    // TODO: this should probably become (AppOrPage.getInitialProps || ssr) in
+    // the next major bump.
+    if (AppOrPage.getInitialProps || !preferStatic) {
```

Comment:
I wonder if now that we're major bumping either way if we want to do the following:

```
if (AppOrPage.getInitialProps || ssr) {}
```

This would mean that only if the `_app` or the root `page` itself has `getInitialProps` OR if the user passes `ssr` as an option this will initiate. This means that the user would have to decide that if one of the page-children has `getInitialProps` they have to add the `ssr` option.

This seems linear with [next-apollo](https://www.npmjs.com/package/next-apollo)

Reply:
I'd be open to that! I like the idea that SSR is now opt-in with static rendering as the default. If you supply a page that defines `getInitialProps`, it'll opt you in to SSR, but if you just implement `getStaticProps` you'll be off to the races w/ SSG. I'm good with it!